### PR TITLE
Give ability to prevent muting of MySQL deprecation warnings

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

I added a `--log-deprecation-warnings` option to the database configuration command that sets a flag on DB configuration.
This flag should allow us to mute warnings related to deprecated SQL features in old migrations, but keep them on fresh install and migrations from 9.5 in order to be able to catch them in our test suite and our development environment.

### CODE REVIEW

**Comment by cedric-anne** [comment]
Example of a failure in install tests : https://github.com/glpi-project/glpi/runs/1734322327?check_suite_focus=true
Example of a failure in update tests : https://github.com/glpi-project/glpi/runs/1733914233?check_suite_focus=true

**Review by trasher — APPROVED** [review]

File: inc/dbconnection.class.php [hunk 1]
```diff
@@ -53,22 +53,24 @@ static function getTypeName($nb = 0) {
...
+    */
+   static function createMainConfig($host, $user, $password, $DBname, $use_utf8mb4 = false, $log_deprecation_warnings = false) {
 
       $DB_str = "<?php\nclass DB extends DBmysql {\n" .
                 "   public \$dbhost      = '$host';\n" .
                 "   public \$dbuser      = '$user';\n" .
                 "   public \$dbpassword  = '". rawurlencode($password) . "';\n" .
                 "   public \$dbdefault   = '$DBname';\n" .
                 "   public \$use_utf8mb4 = " . ($use_utf8mb4 ? 'true' : 'false') . ";\n" .
+                "   public \$log_deprecation_warnings = " . ($log_deprecation_warnings ? 'true' : 'false') . ";\n" .
```
Comment (orthagh) [inline inc/dbconnection.class.php:73]: As the property is present in the parent, could we write the line only when `$log_deprecation_warnings` is present.
I'm not sure the parameter will be usefull in production instances (Is the purpose of this is only catching with CI on our side ?) and set it with false default value will only bring questions on support.

Comment (cedric-anne) [reply inc/dbconnection.class.php]: Indeed, no need to get support due to this.

While changing it, I take time to add unit tests on this, so I had to made some code refactoring.

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]
Please rebase to solve conflict

**Pull request merged**